business/apps_test.go

```diff
@@ -4,54 +4,72 @@ import (
 	"context"
 	"testing"
 
-	osapps_v1 "github.com/openshift/api/apps/v1"
 	osproject_v1 "github.com/openshift/api/project/v1"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
-	apps_v1 "k8s.io/api/apps/v1"
-	batch_v1 "k8s.io/api/batch/v1"
 	core_v1 "k8s.io/api/core/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 
 	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/kubernetes/cache"
 	"github.com/kiali/kiali/kubernetes/kubetest"
 	"github.com/kiali/kiali/prometheus/prometheustest"
 )
 
-func setupAppService(k8s *kubetest.K8SClientMock) *AppService {
+func setupAppService(k8s *kubetest.FakeK8sClient) *AppService {
 	prom := new(prometheustest.PromClientMock)
 	layer := NewWithBackends(k8s, prom, nil)
 	setupGlobalMeshConfig()
 	return &AppService{k8s: k8s, prom: prom, businessLayer: layer}
 }
 
+func setupTestingKialiCache(k8s *kubetest.FakeK8sClient, conf *config.Config, require *require.Assertions) func() {
```

Comment:
Not sure, if this can be reused for other tests and might be placed as a common utility?

Reply:
Quite possible. I was thinking in moving this to a shared place in a follow-up PR, if needed. But I think I won't do more work on the tests. Let me check where I can move this and change this PR.

Reply:
I think it is Ok to change it in a follow up PR